documentation/general/analyzer-redirecting.md

```diff
@@ -0,0 +1,129 @@
+  Now we will redirect the analyzers to those from SDK 9.0.1 that are deployed as part of VS and reference the matching Roslyn (4.12).
+
+Loading analyzers with older major version should not be a problem because they must reference an older version of Roslyn.
+
+Loading analyzers with newer major version in an old VS will result in an error like:
+
+> error NETSDK1045: The current .NET SDK does not support targeting .NET 10.0.  Either target .NET 9.0 or lower, or use a version of the .NET SDK that supports .NET 10.0. Download the .NET SDK from
+https://aka.ms/dotnet/download
+
+## Overview
+
+- The SDK will contain [a mapping file][file-format] listing the analyzers that want to [dual-insert][dual-insert] into VS.
+
+- Those analyzers will be deployed with VS in some `{VS-analyzers}` folder
+  (to be determined, could be something like `C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\DotNetAnalyzers`).
+
+  - This will happen as part of the SDK being inserted into VS.
+  
+  - The mapping file [will be used][vs-insertions] to automatically gather all the analyzers.
+
+  - The mapping file will be deployed there as well (at `{VS-analyzers}\mapping.txt`) so Roslyn can read it.
+
+    - If multiple SDKs are inserted into VS, the most recent version wins.
+      The mapping file should be backwards-compatible (we should not remove patterns from it).
+
+- Roslyn [will use][roslyn-redirecting] the mapping file to redirect analyzer loads from SDK to VS.
+
+  - If the file is not found at `{VS-analyzers}\mapping.txt`, analyzer loading continues normally from SDK.
+
+  - If a DLL load is requested which matches a mapping, it is redirected via the [IAnalyzerAssemblyResolver][analyzer-assembly-resolver] infrastructure.
+
+## Mapping file format
+[file-format]: #mapping-file-format
+
+The mapping file v1 is a simple list of directory "patterns". For example:
+
+```
+packs/Microsoft.NETCore.App.Ref/*/analyzers/dotnet/cs
+sdk/*/Sdks/Microsoft.NET.Sdk/analyzers
+sdk/*/Sdks/Microsoft.NET.Sdk.Web/analyzers/cs
+```
+
+Each pattern can contain a single asterisk `*` where a version is expected.
+
+Lines starting with `#` are ignored (can be used as comments).
+
+### Inserting analyzers into VS
+[vs-insertions]: #inserting-analyzers-into-vs
+
+The file is used to copy analyzers into VS.
+The paths start at the dotnet root.
+Only the major version from the `*` part is preserved.
+Rest of the path is preserved in the destination.
+So for example given the following pattern
+
+```
+packs/Microsoft.NETCore.App.Ref/*/analyzers
+```
+
+we copy
+
+```
+{dotnet-root}\packs\Microsoft.NETCore.App.Ref\8.0.7\analyzers\**
+```
+
+into
+
+```
+{VS-analyzers}\packs\Microsoft.NETCore.App.Ref\8\analyzers\**
+```
+
+If multiple SDKs are inserted into VS,
+they should have different major versions,
+but if there are two SDKs trying to copy the same pattern with the same major version,
+only the more recent SDK should win.
```

Comment:
Hmm ... I wonder if we should error to make sure there isn't an accidental conflict. Basically the assumption being that we shouldn't be inserting the same analyzer version from two different SDKs so error if it happens. If it happens and there is a legitimate use case then we can re-evaluate.